client/state/comments/reducer.js

```diff
@@ -190,15 +196,24 @@ export const totalCommentsCount = createReducer(
 export const errors = createReducer(
 	{},
 	{
-		[ COMMENTS_ERROR ]: ( state, action ) => ( {
-			...state,
-			[ `${ action.siteId }-${ action.commentId }` ]: { error: true },
-		} ),
+		[ COMMENTS_ERROR ]: ( state, action ) => {
```

Comment:
Transient 500 should have been taken care of by automatic retries. 

As the reducer is written now, a hard refresh would be necessary.  But, note, this is only about errors loading a comment by id. We fallback to regular comment loading